docker/mongodb-kubernetes-tests/tests/authentication/replica_set_oidc_m2m_group.py

```diff
@@ -0,0 +1,145 @@
+@pytest.mark.e2e_replica_set_oidc_m2m_group
+class TestAddNewOIDCProviderAndRole(KubernetesTester):
+    def test_add_oidc_provider_and_role(self, replica_set: MongoDB):
+        replica_set.assert_reaches_phase(Phase.Running, timeout=400)
+
+        replica_set.load()
+
+        new_oidc_provider_config = {
+            "audience": "dummy-audience",
+            "issuerURI": "https://valid-issuer.example.com",
+            "requestedScopes": [],
+            "userClaim": "sub",
+            "groupsClaim": "group",
+            "authorizationMethod": "WorkloadIdentityFederation",
+            "authorizationType": "GroupMembership",
+            "configurationName": "dummy-oidc-config",
+        }
+
+        new_role = {
+            "role": "dummy-oidc-config/test",
+            "db": "admin",
+            "roles": [{"role": "readWriteAnyDatabase", "db": "admin"}],
+        }
+
+        replica_set.append_oidc_provider_config(new_oidc_provider_config)
+        replica_set.append_role(new_role)
+
+        replica_set.update()
+
+        def config_and_roles_preserved() -> bool:
+            tester = replica_set.get_automation_config_tester()
+            try:
+
+                tester.assert_authentication_mechanism_enabled("MONGODB-OIDC", active_auth_mechanism=False)
+                tester.assert_authentication_enabled(2)
+                tester.assert_expected_users(0)
+                tester.assert_has_expected_number_of_roles(expected_roles=2)
+
+                expected_oidc_configs = [
+                    {
+                        "audience": oidc.get_cognito_workload_client_id(),
+                        "issuerUri": oidc.get_cognito_workload_url(),
+                        "clientId": oidc.get_cognito_workload_client_id(),
+                        "userClaim": "sub",
+                        "groupsClaim": "cognito:groups",
+                        "JWKSPollSecs": 0,
+                        "authNamePrefix": "OIDC-test",
+                        "supportsHumanFlows": False,
+                        "useAuthorizationClaim": True,
+                    },
+                    {
+                        "audience": "dummy-audience",
+                        "issuerUri": "https://valid-issuer.example.com",
+                        "userClaim": "sub",
+                        "groupsClaim": "group",
+                        "JWKSPollSecs": 0,
+                        "authNamePrefix": "dummy-oidc-config",
+                        "supportsHumanFlows": False,
+                        "useAuthorizationClaim": True,
+                    },
+                ]
+
+                tester.assert_oidc_configuration(expected_oidc_configs)
+                return True
+            except AssertionError:
+                return False
+
+        wait_until(config_and_roles_preserved, timeout=300, sleep=5)
+
+        replica_set.assert_reaches_phase(Phase.Running, timeout=400)
```

Comment:
shouldn't we first wait for Running phase and then check the oidc configs? If we do it other way than we cannot guarantee that Running state will have unchanged oidc config